agent/engine/errors.go

```diff
@@ -20,7 +20,11 @@ import (
 	"github.com/aws/amazon-ecs-agent/agent/engine/dockerclient"
 )
 
-const dockerTimeoutErrorName = "DockerTimeoutError"
+const (
+	dockerTimeoutErrorName     = "DockerTimeoutError"
+	unretriableDockerErrorName = "UnretriableDockerError"
```

Comment:
These new error names will show up in our APIs.  Do we want to do that, or should we make them show up as `CannotStopContainerError`?